src/execute/update_access_routes.rs

```diff
@@ -0,0 +1,547 @@
+use crate::util::contract_helpers::check_funds_are_empty;
+use crate::util::event_attributes::{EventAttributes, EventType};
+use crate::util::functions::filter_valid_access_routes;
+use crate::util::traits::ResultExtensions;
+use cosmwasm_std::{MessageInfo, Response};
+
+#[derive(Clone, PartialEq)]
+pub struct UpdateAccessRoutesV1 {
+    pub identifier: AssetIdentifier,
+    pub owner_address: String,
+    pub access_routes: Vec<AccessRoute>,
+}
+impl UpdateAccessRoutesV1 {
+    pub fn new<S: Into<String>>(
+        identifier: AssetIdentifier,
+        owner_address: S,
+        access_routes: Vec<AccessRoute>,
+    ) -> Self {
+        Self {
+            identifier,
+            owner_address: owner_address.into(),
+            access_routes,
+        }
+    }
+
+    pub fn from_execute_msg(msg: ExecuteMsg) -> AssetResult<Self> {
+        match msg {
+            ExecuteMsg::UpdateAccessRoutes {
+                identifier,
+                owner_address,
+                access_routes,
+            } => Self::new(identifier, owner_address, access_routes).to_ok(),
+            _ => ContractError::InvalidMessageType {
+                expected_message_type: "ExecuteMsg::UpdateAccessRoutes".to_string(),
+            }
+            .to_err(),
+        }
+    }
+}
+
+pub fn update_access_routes<'a, T>(
+    repository: T,
+    info: MessageInfo,
+    msg: UpdateAccessRoutesV1,
+) -> EntryPointResponse
+where
+    T: AssetMetaRepository + MessageGatheringService + DepsManager<'a>,
+{
+    check_funds_are_empty(&info)?;
+    // If the sender is not the specified owner address and the sender is not the admin, they are
+    // not authorized to change access routes
+    if info.sender != msg.owner_address
+        && info.sender
+            != repository
+                .use_deps(|deps| config_read_v2(deps.storage).load())?
+                .admin
+    {
+        return ContractError::Unauthorized {
+            explanation:
+                "only the admin or owner of the given access routes can make modifications to them"
+                    .to_string(),
+        }
+        .to_err();
+    }
+    let mut access_routes = filter_valid_access_routes(msg.access_routes.clone());
+    if msg.access_routes.len() != access_routes.len() {
+        // The filtration function will trim duplicate routes, as well as invalid routes
+        return ContractError::generic("invalid or duplicate access routes were provided").to_err();
+    }
+    let scope_address = msg.identifier.get_scope_address()?;
+    let mut scope_attribute = repository.get_asset(&scope_address)?;
+    if let Some(mut target_access_definition) = scope_attribute
+        .access_definitions
+        .iter()
+        .find(|def| def.owner_address == msg.owner_address)
+        .map(|def| def.to_owned())
+    {
+        // Filter the access definition to be changed from the attribute's vector
+        scope_attribute.access_definitions = scope_attribute
```

Comment:
Or, as they say in Kotlin:
```kotlin
scopeAttribute.accessDefinitions.filter { it.ownerAddress != targetAccessDefinition.ownerAddress }.plus(targetAccessDefinition)
```

But this is Rust, so it's gotta be 12 lines instead 🥇